### DISCUSSION TITLE
Problem getting started with 2016 MSSEG dataset

### DISCUSSION THREAD

**Discussion by savvvyk** [Project-MONAI/MONAI#3767] [Q&A] [answered]
Hello,

I'm modifying the spleen 3D segmentation example to work with the 2016 MSSEG dataset.  The dataset consists of multiple input images (I'm using 3)  and a single consensus label image.  I'm currently using the pre-processed images but without the registered brain mask file.  I'm currently experiencing two issues: first is my label image is shows as blank after the data loader but doesn't error (minor issue) and the second is AssertionError: ground truth has different shape (major issue, full error below). I would appreciate any help or suggestions.  I've checked that onehot is set to true and I've tried remapping the labels from [1,2] to [0,1] but neither has made a difference.

epoch 1/600
/usr/local/lib/python3.7/dist-packages/monai/losses/dice.py:135: UserWarning: single channel prediction, `softmax=True` ignored.
  warnings.warn("single channel prediction, `softmax=True` ignored.")
/usr/local/lib/python3.7/dist-packages/monai/losses/dice.py:144: UserWarning: single channel prediction, `to_onehot_y=True` ignored.
  warnings.warn("single channel prediction, `to_onehot_y=True` ignored.")

AssertionError                            Traceback (most recent call last)
<ipython-input-17-f155a20833f8> in <module>()
     22         optimizer.zero_grad()
     23         outputs = model(inputs)
---> 24         loss = loss_function(outputs, labels)
     25         loss.backward()
     26         optimizer.step()

1 frames
/usr/local/lib/python3.7/dist-packages/monai/losses/dice.py in forward(self, input, target)
    155 
    156         if target.shape != input.shape:
--> 157             raise AssertionError(f"ground truth has different shape ({target.shape}) from input ({input.shape})")
    158 
    159         # reducing only spatial dimensions (not batch nor channels)

AssertionError: ground truth has different shape (torch.Size([1, 128, 64, 128, 128])) from input (torch.Size([1, 1, 64, 128, 128]))

**Comment by Nic-Ma** [comment]
Hi @savvvyk ,

Thanks for your interest here.
For the first look at your error message, I feel it's related to your data shape, seems it's not correct.
Could you please help change all your `AddChanneld` transform to `EnsureChannelFirstd` transform and try again?

Thanks in advance.

**Reply by savvvyk** [reply]
@Nic-Ma, Thank you for your quick response.  I tried switching `AddChanneld `for `EnsureChannelFirstd `and observed the same ground truth shape issue as before.  The linked code file should be updated to reflect the latest.

**Comment by savvvyk** [comment, marked as answer]
I discovered that the label was not being included in the EnsureChannelFirstd command causing it to not have the channel array element.  I have now been able to run the code without error (still needs much tuning however).
